spec/adapters/threema_adapter/outbound_spec.rb

```diff
@@ -24,29 +23,40 @@
 
   describe '::send_welcome_message!' do
     subject { -> { described_class.send_welcome_message!(contributor) } }
+
     it { should enqueue_job(described_class) }
+
     context 'contributor has no threema_id' do
       let(:contributor) { create(:contributor, threema_id: nil, email: nil) }
       it { should_not enqueue_job(described_class) }
     end
   end
 
   describe '#perform' do
-    subject { adapter.perform(text: message.text, recipient: message.recipient) }
+    subject { -> { adapter.perform(text: message.text, recipient: message.recipient) } }
 
     it 'sends the message' do
-      expect(threema).to receive(:send).with({ type: :text, threema_id: contributor.threema_id, text: message.text })
-
-      subject
+      expect_any_instance_of(Threema).to receive(:send).with(type: :text, threema_id: 'V5EA564T', text: message.text)
+      subject.call
     end
 
-    describe '#welcome_message' do
-      subject { described_class.welcome_message }
-      it 'strips whitespace to not break basic formatting' do
-        allow(Setting).to receive(:onboarding_success_heading).and_return(" \n text with leading and trailing whitespace \t \n ")
-        allow(Setting).to receive(:onboarding_success_text).and_return("\nSuccess text.\n")
-        is_expected.to eq("*text with leading and trailing whitespace*\n\nSuccess text.\n")
+    context 'with lowercase Threema ID' do
+      let(:threema_id) { 'v5ea564t' }
+
+      it 'converts ID to uppercase' do
+        expect_any_instance_of(Threema).to receive(:send).with(type: :text, threema_id: 'V5EA564T', text: message.text)
+        subject.call
       end
     end
   end
+
+  describe '#welcome_message' do
+    subject { described_class.welcome_message }
+
+    it 'strips whitespace to not break basic formatting' do
+      allow(Setting).to receive(:onboarding_success_heading).and_return(" \n text with leading and trailing whitespace \t \n ")
+      allow(Setting).to receive(:onboarding_success_text).and_return("\nSuccess text.\n")
+      is_expected.to eq("*text with leading and trailing whitespace*\n\nSuccess text.\n")
+    end
+  end
```

Comment:
Is this test block unrelated? It seems to me that we didn't ask @roschaefer to add a test for this in the `PR` that introduced it so we are sneaking it in here?

If so, I generally prefer to avoid this and create a separate micro PR whose objective is to add missing tests, but am open to discussing this with the team.

Reply:
Oh, sorry... When I apply `Hide whitespaces` I can see that these are not changes you added... :blush: